usr/share/rear/lib/network-functions.sh

```diff
@@ -723,3 +723,28 @@ is_bonding_device ()
    [ -f "/sys/class/net/$1/bonding/slaves" ]
 }
 
+#  ^(([1-9]|[12]?[0-9]{1,2}|2([0-4][0-9]|5[0-5]))\.){3}([1-9]|[12]?[0-9]{1,2}|2([0-4][0-9]|5[0-5]))$
+
+# Retun 0 if args is a valid IPV4 ip_address
+function is_ip() {
+    local test_ip=$1
+    [ -z $test_ip ] && BugError "function is_ip() called without argument."
+
+    # test if $test_ip is a valide IPV4 address: "[1 to 255].[1 to 255].[1 to 255].[1 to 255]"
+    if [[ $test_ip =~ ^(([0-9]{1,2}|1[0-9]{2}|2([0-4][0-9]|5[0-5]))\.){3}([0-9]{1,2}|1[0-9]{2}|2([0-4][0-9]|5[0-5]))$ ]] ; then
```

Comment:
It seems to me like `ping 127.0.0.1.` also works so you could check for `[1 to 255].` 4 times. Another approach would be to just check the string for containing only digits and dots like `[[ "$test_ip" =~ ^[0-9.]+$ ]]`

And please quote the `$test_ip` parameter everywhere to protect against wrong calls with a blank in it.

Reply:
for this one : `[ -z $test_ip ] && BugError` I would prefer to not use `"` Because I don't want to consider a space `" "` as a valid parameter.

I will update the other one.

Reply:
Giving a space will fail the IP test so that is fine. The BugError only makes sure that people always do give some parameter. Try calling your code with `"hello world"` and you will see that it fails not as intended.